packages/devtools/src/common/promise.ts

```diff
@@ -72,3 +73,131 @@ export const firstFactory =
     <TInput extends unknown[], TOutput>(...factories: Factory<TInput, TOutput>[]): Factory<TInput, TOutput> =>
     async (...input) =>
         await first(factories.map((factory) => () => factory(...input)))
+
+/**
+ * RetryStrategy is a way of having a fine grain control
```

Comment:
nit: RetryStrategy does this, but it is not that, what is it? I guess I am asking about the type or pattern in which it fits.

Reply:
`RetryStrategy` is a type that represents a function. This function with its signature is used in `createRetryFactory` to control how and if retrying will be performed.